data/entities.yaml

```diff
@@ -137,6 +137,50 @@
     water or steam.
   properties: [portable]
 
+- name: iron plate
+  display:
+    attr: iron
+    char: '■'
+  description:
+  - Raw iron ore. Used to create more resilient tools.
+  - It can only be mined by drilling in the mountains.
+  properties: [portable]
+
+- name: iron gear
+  display:
+    attr: iron
+    char: '*'
+  description:
+  - An iron gear.
+  properties: [portable]
+
+- name: iron ore
+  display:
+    attr: copper
+    char: 'C'
+  description:
+  - Raw iron ore. Used to create more resilient tools.
+  - It can only be mined by drilling in the mountains.
+  properties: [portable]
+
+- name: iron vein
+  display:
+    attr: iron'
+    char: 'A'
+  description:
+  - A place in the mountains where raw iron ore can be mined.
+    As it is hidden inside a mountain, a tunnel needs to be
+    first drilled through, so that the vein becomes accessible.
+  properties: [unwalkable]
+
+- name: iron mine
```

Comment:
So one idea I had was to get rid of the distinction between `copper vein`/`iron vein` and `copper mine`/`iron mine` and just have `vein` and `mine`.  Given #372, the idea is that drilling in a mine would give you copper with some probability and iron with another (smaller) probability. We can also easily do things like a wooden drill might get you copper but never iron, whereas an iron drill could get you both, etc. --- we just have to specify the right recipes. While this scenario of getting everything from a single mine may not be very accurate in the "real world", I like it for a few reasons:

- It keeps things simpler in terms of the number of different entities in the world.
- It leaves the door open for adding other mine-able things (gold? diamonds? etc.) with much lower probabilities, without having to add yet more separate types of veins and mines.
- With multiple different types of mines, the difficulty for a player lies in locating all the different types and building separate drilling operations at each. With a single mine, the difficulty for the player lies in building a drilling and processing/sorting operation.  The latter sounds much more fun and interesting from a programming perspective than the former.

What do you think?

Reply:
I actually like both ideas - maybe we can split it into different scenarios :thinking: 

It does not require any extra logic in Swarm so the only overhead is extra recipes in our main `data/recipes.yaml`.
I think moving it to some *adventure mode* scenario would both clean up and leave our options open.

My only concern is how to order the changes - I will have to check if scenarios support all I need now or if it is easier to move it over later. :slightly_smiling_face:

Reply:
Alternatively, we could parametrize entities by their underlying material to cut down on boilerplate.

I am not sure how it would work, but maybe it will become more evident once we add gold and other late-game stuff.

Reply:
You said you like both ideas.  What is it you like about the idea of having separate `copper mine`, `iron mine`, etc.?

I don't mind supporting both, and I am not so much concerned about boilerplate as I am about the gameplay experience.  I don't mind having extra entities and recipes in `recipes.yaml` etc.  So we could have both `mine` that works as I described with weighted random results as well as specific `iron mine`, `copper mine`, etc.  The biggest issues then would be (1) which version do you get in the "default" mode?  Even if we support both we still have to decide which is the default game.  And (2) in order to support both we have to allow multiple "world functions" and provide a mechanism for scenarios to specify which world function they will use.  I think this was discussed in an issue somewhere but I can't find it at the moment.

Reply:
I like that it is visual, easier to debug and requires sending robots a bit further. :slightly_smiling_face: 

We could actually do with only mines and have them randomly drilled from `mountain`. That would address the number of entities in the world and make it simpler to add new mines, but it would be even harder to establish mining operations.

Reply:
Hmm, I like the idea of mines appearing from a drilled mountain with a certain probability (in all cases, whether there is only one kind of mine or multiple).  It requires more exploration.  It also feels weird to me that right now the mountains are different colors depending on whether there is a vein --- not that Swarm needs to reflect the real world necessarily, but if you have to drill for something, you should not be able to tell just by looking at the surface what you will find when you drill.

Reply:
I think I found a good balance:
- mountains can be drilled with an 8/1/1 probability to tunnel/copper mine/iron mine
- in scenarios, we can use the deterministic vein ➔ mine custom recipes and entities

Reply:
Ah, OK, I like that.